tryoutmanager/models.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3.6
+
+import os
+import uuid
+import sqlalchemy as sa
+
+from datetime import datetime
+from sqlalchemy import Column, String, Boolean, DateTime, Table, \
+                       ForeignKey, Enum, Index, UniqueConstraint
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.ext.declarative import declarative_base
+
+db = sa.create_engine(os.environ['DATABASE_URL'])
+metadata = sa.MetaData()
+
+Base = declarative_base(metadata)
+Session = sessionmaker(bind=db)
+
+
+# Association tables for many-to-many relationships
+markers_challenges = Table('markers_challenges',
+                           Base.metadata,
+                           Column('marker_email',
+                                  String,
+                                  ForeignKey('markers.email')),
+                           Column('challenge_id',
+                                  UUID,
+                                  ForeignKey('challenges.id')))
+
+challenges_submissions = Table('challenges_submissions',
+                               Base.metadata,
+                               Column('challenge_id',
+                                      UUID,
+                                      ForeignKey('challenges.id')),
+                               Column('submission_id',
+                                      UUID,
+                                      ForeignKey('submissions.id')))
+
+
+# Tables corresponding to concrete models
+class AdminUser(Base):
```

Comment:
Do we specifically need an AdminUser which holds everything separate from a normal user?

Reply:
Given that we had previously discussed having two admin tiers (challenge markers with authority over submissions to challenges they mark) and superuser-type admins, and how different these roles are, I think having a separate `AdminUser` model makes sense.

Reply:
Interesting, I think we can work with this.